### PR TITLE
minor adjustments + add Current Weather Card component

### PR DESCRIPTION
**Pull request by Lvlad27**
minor adjustments + add Current Weather Card component

image

### CODE REVIEW

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import styles from './CurrWeatherCard.module.scss';
+import endpoints from '../../Helpers/endpoints';
+
+export default function CurrWeatherCard(props) {
+    let locationData = JSON.parse(props.locationData);
+    let currWeather = JSON.parse(props.currWeather);
```
Comment (JuliaShavyrova) [inline src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: would be better to pass already transformed data through props

File: src/Components/Navbar/Navbar.module.scss [hunk 2]
```diff
@@ -1,24 +1,44 @@
...
+    margin-bottom: 2rem;
+    backdrop-filter: blur(16px) saturate(180%);
+    -webkit-backdrop-filter: blur(16px) saturate(180%);
+    background-color: rgba(255, 255, 255, 0.75);
+    border: 1px solid rgba(209, 213, 219, 0.3);
+
+    //   --mask: linear-gradient(#000 0 0) center/calc(100% - 27.75px) calc(100% - 27.75px) no-repeat,
+    //     radial-gradient(farthest-side, #000 98%, #0000) 0 0/30px 30px round;
+    //   -webkit-mask: var(--mask);
+    //   mask: var(--mask);
```
Comment (JuliaShavyrova) [inline src/Components/Navbar/Navbar.module.scss]: please clean code

File: src/Pages/Home/Home.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    async function getCurrentWeather() {
+        if (!coords) {
+            return;
+        }
+        try {
+            setIsLoading(true);
+            const { data } = await instance.get(endpoints.CURR_WEATHER(coords));
+            setCurrWeather(data);
+            setIsLoading(false);
```
Comment (JuliaShavyrova) [inline src/Pages/Home/Home.jsx]: will user see infinite "loading..." in case the request failed?

File: src/Pages/Home/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        setIsLoading(false);
+    }, [coords]);
+
+    return (
+        <div className={styles.container}>
+            {!isLoading && locationData && (
+                <CurrWeatherCard locationData={locationData} currWeather={currWeather} />
+            )}
+            {!isLoading && !locationData && <p>Loading...</p>}
+            {isLoading && locationData && <p>Loading...</p>}
```
Comment (JuliaShavyrova) [inline src/Pages/Home/Home.jsx]: conditions can be simplified and combined

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx (hunk 1 above)
Comment (Lvlad27) [reply src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: done

File: src/Components/Navbar/Navbar.module.scss (hunk 2 above)
Comment (Lvlad27) [reply src/Components/Navbar/Navbar.module.scss]: done

File: src/Pages/Home/Home.jsx (hunk 3 above)
Comment (Lvlad27) [reply src/Pages/Home/Home.jsx]: Fixed by setting error state.

File: src/Pages/Home/Home.jsx (hunk 4 above)
Comment (Lvlad27) [reply src/Pages/Home/Home.jsx]: done

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import styles from './CurrWeatherCard.module.scss';
+import endpoints from '../../Helpers/endpoints';
+
+export default function CurrWeatherCard(props) {
+  let locationData = props.locationData;
+  let currWeather = props.currWeather;
```
Comment (JuliaShavyrova) [inline src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: props can be destructured so no need to create new variables

File: src/DataService/cookieService.js [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
+  document.cookie = `${key}=${value}`;
 }
 
 function loadCookie(key) {
-    let cookie = document.cookie;
-    cookie = cookie.substring(cookie.indexOf('=') + 1);
-    return cookie;
+  let cookie = document.cookie;
+  cookie = cookie.substring(cookie.indexOf('=') + 1);
+  return cookie;
```
Comment (JuliaShavyrova) [inline src/DataService/cookieService.js]: a bit of improvement
```suggestion
  const cookie = document.cookie;
  return cookie.substring(cookie.indexOf('=') + 1);;
```

File: src/Pages/Home/Home.jsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  async function getLocationData() {
+    if (!coords) {
+      return;
+    }
+    try {
+      setIsLoading(true);
+      let { data } = await instance.get(endpoints.LOCATION_DATA(coords));
+      setLocationData(JSON.parse(data));
```
Comment (JuliaShavyrova) [inline src/Pages/Home/Home.jsx]: why not using axios config to transform requests data?

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx (hunk 5 above)
Comment (Lvlad27) [reply src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: done

File: src/DataService/cookieService.js (hunk 6 above)
Comment (Lvlad27) [reply src/DataService/cookieService.js]: done

File: src/Pages/Home/Home.jsx (hunk 7 above)
Comment (Lvlad27) [reply src/Pages/Home/Home.jsx]: done

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**